### PR TITLE
Extract team location to autopopulate hospitals in surrounding area for sample collection 

### LINKED ISSUE #349: Extract team location to autopopulate hospitals in surrounding area for sample collection
- [ ] Prepopulate names based on user location if available, make it an editable field

- Pulldown menus are preferred, however, prepopulated field is suffice
- Pulldown menus are done in Reveal
- Another thing we can explore is a hierarchical view, like reveal, locations and publications. If the doctors generally move around those health facilities, we can set these in the dropdown and repopulate them dynamically.
- Easier to repopulate it with team locations though which are stored in OpenMRS and locations can be tracked down by GPS

### PR DESCRIPTION
**Pull request by syedowaisali90**
Hi, @vincent-karuri I have created a PR for review.
Close #349

### CODE REVIEW

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 1]
```diff
@@ -117,6 +119,9 @@ protected void prePopulateRDTPatientFields(Patient patient, JSONObject field) th
         } else if (PATIENT_AGE.equals(key)) {
             field.put(JsonFormUtils.VALUE, patient.getAge());
         }
+        else if (CovidConstants.FormFields.FACILITY_NAME.equals(key) || CovidConstants.FormFields.HEALTH_FACILITY_NAME.equals(key) || CovidConstants.FormFields.OTHER_HEALTH_FACILITY_TYPE_NAME.equals(key)) {
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:122]: Any reason for the 3 checks? Are there multiple fields on the RDT form that require to be pre-populated with the locations or is it just the `CovidConstants.FormFields.FACILITY_NAME`.

Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:122]: https://github.com/onaio/rdt-standard/issues/349#issuecomment-701148554 this only populates the RDT test form. `Symptom and history form` and `Sample collection form` will also need population.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 2]
```diff
@@ -147,4 +152,15 @@ private String getLoggedInUserPreferredName() {
...
+    private JSONArray getLocations() {
+        JSONArray jsonArray = new JSONArray();
+        AllSharedPreferences allSharedPreferences = RDTApplication.getInstance().getContext().allSharedPreferences();
+        String defaultLocationUuid = allSharedPreferences.fetchDefaultLocalityId(allSharedPreferences.fetchRegisteredANM());
+        List<String> locations = LocationHelper.getInstance().getOpenMrsLocationHierarchy(defaultLocationUuid, false);
+        for (String location : locations) {
+            jsonArray.put(location);
+        }
+        return jsonArray;
+    }
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:165]: Is this logic verified? It looks like it pulls the entire location hierarchy and not the sibling locations as described here https://github.com/onaio/rdt-standard/issues/349#issuecomment-700727776.

**Comment by vincent-karuri** [comment]
Could we add unit tests for the pre-population of location logic? Specifically, verify that the correct (expected) locations are populated to the three forms.

File: rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json [hunk 3]
```diff
@@ -88,22 +88,18 @@
...
         "openmrs_entity_id": "",
-        "type": "edit_text",
+        "type": "spinner",
+        "hint": "Name of facility",
         "v_required": {
           "value": "false",
           "err": ""
-        }
+        },
+        "values": []
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json:102]: The spinner here and in all other instances should follow the [new spinner format](https://github.com/OpenSRP/opensrp-client-native-form/wiki/Spinner-Field#spinner-20-mls-20-compatible) that uses an options block.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 1 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:122]: yeah, we need to implement this feature in 3 different forms because all three forms have a different field name. Therefore i have used to OR check.

Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:122]: it works in all three forms

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 2 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:165]: I have checked this method it is returning only one location Indonesia. I have also tried other child-related methods but all of them returning null.

Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:165]: Actually, I can't determine whats the exact method will be used for that. Most of the method which I have tried that is returning null or not record.

File: rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json (hunk 3 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json:102]: fixed.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 1 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:122]: Ok, could we move the fields into a set and create a private method that checks if the key is part of that set? Similar to https://github.com/onaio/rdt-standard/blob/e71190b6976212242a686fbb9aa6e84a9ea4a7b0/rdt-open-reader/src/main/java/io/ona/rdt/util/RDTJsonFormUtils.java#L80.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 2 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:165]: I changed the location hierarchy to have a few more levels. Could you verify that more locations show up in the hierarchy?

The user `indtester1` should be assigned to `Indonesia Location 3`. The dropdown should show `Indonesia Location 1, 2 and 3`.

File: rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json (hunk 3 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json:102]: I can't find the fix for this. Did you push it?

Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form-original.json:102]: I have fixed that locally but didn't push the changes

File: rdt-open-reader/src/main/java/io/ona/rdt/job/RDTJobCreator.java [hunk 4]
```diff
@@ -34,6 +37,8 @@ public Job create(@NonNull String tag) {
                 return new RDTSyncSettingsServiceJob();
             case LocationStructureServiceJob.TAG:
                 return new LocationStructureServiceJob();
+            case SyncAllLocationsServiceJob.TAG:
+                return new SyncAllLocationsServiceJob(RDTSyncAllLocationsIntentService.class);
```
Comment (vincent-karuri) [inline rdt-open-reader/src/main/java/io/ona/rdt/job/RDTJobCreator.java:41]: `SyncAllLocationsServiceJob` should be overridden, it does not run `RDTSyncAllLocationsIntentService` and instead runs `SyncAllLocationsIntentService` which is not what we want.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 5]
```diff
@@ -147,4 +163,40 @@ private String getLoggedInUserPreferredName() {
         final AllSharedPreferences allSharedPreference = RDTApplication.getInstance().getContext().allSharedPreferences();
         return allSharedPreference.getANMPreferredName(allSharedPreference.fetchRegisteredANM());
     }
+
+    private JSONArray getLocations() throws JSONException {
+        JSONArray jsonArray = new JSONArray();
+        AllSharedPreferences allSharedPreferences = RDTApplication.getInstance().getContext().allSharedPreferences();
+        String defaultLocationUuid = allSharedPreferences.fetchDefaultLocalityId(allSharedPreferences.fetchRegisteredANM());
+        LinkedHashMap<String, TreeNode<String, Location>> locationMap = LocationHelper.getInstance().map();
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:171]: Isn't this starting from the root location? The reason for enabling `RDTSyncAllLocationsIntentService` is to be able to start traversal from the immediate parent location.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 6]
```diff
@@ -147,4 +163,40 @@ private String getLoggedInUserPreferredName() {
...
+                return locations;
+            }
+            else {
+                return filterLocations(locationId, entry.getValue().getChildren());
+            }
+        }
+        else {
+            return new ArrayList<>();
+        }
+    }
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:201]: This logic can be simplified a bit by removing the external if/else block.

File: rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java [hunk 7]
```diff
@@ -161,4 +174,28 @@ public static boolean tableExists(SQLiteDatabase db, String tableName) {
...
+                return entry.getValue().getParent();
+            }
+            else {
+                return getParentLocationId(childLocationId, entry.getValue().getChildren());
+            }
+        }
+        else {
+            return null;
+        }
+    }
```
Comment (vincent-karuri) [inline rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java:200]: Can this https://github.com/OpenSRP/opensrp-client-core/blob/80318160b7a0194e2c1e706438358e39f71bd9b2/opensrp-app/src/main/java/org/smartregister/location/helper/LocationHelper.java#L588  and this https://github.com/OpenSRP/opensrp-client-core/blob/80318160b7a0194e2c1e706438358e39f71bd9b2/opensrp-app/src/main/java/org/smartregister/location/helper/LocationHelper.java#L596 be used to get the parent location id?

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 8]
```diff
@@ -147,4 +163,40 @@ private String getLoggedInUserPreferredName() {
...
+                }
+                return locations;
+            }
+            else {
+                return filterLocations(locationId, entry.getValue().getChildren());
+            }
+        }
+        else {
+            return new ArrayList<>();
+        }
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:200]: This logic can be simplified a bit by removing the external if/else block.

File: rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java (hunk 7 above)
Comment (vincent-karuri) [inline rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java:200]: Can this https://github.com/OpenSRP/opensrp-client-core/blob/80318160b7a0194e2c1e706438358e39f71bd9b2/opensrp-app/src/main/java/org/smartregister/location/helper/LocationHelper.java#L588  and this https://github.com/OpenSRP/opensrp-client-core/blob/80318160b7a0194e2c1e706438358e39f71bd9b2/opensrp-app/src/main/java/org/smartregister/location/helper/LocationHelper.java#L596 be used to get the parent location id?

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 5 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:171]: it's starting from the root location by using `map()` and traverse to the immediate parent location by the given child location id in a param and get all the child locations from the immediate parent by the given child location id.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 8 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:200]: fixed.

File: rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java (hunk 7 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java:200]: fixed.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 8 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:200]: Also, we may not need this recursion if we start from the immediate parent location.

**Comment by vincent-karuri** [comment]
The app crashes when launching the forms. Please verify that this feature works.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/job/RDTSyncAllLocationsServiceJob.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import androidx.annotation.NonNull;
+
+import com.evernote.android.job.Job;
+
+import org.smartregister.job.SyncAllLocationsServiceJob;
+import org.smartregister.sync.intent.SyncAllLocationsIntentService;
+
+import io.ona.rdt.application.RDTApplication;
+
+public class RDTSyncAllLocationsServiceJob extends SyncAllLocationsServiceJob {
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/job/RDTSyncAllLocationsServiceJob.java:14]: Why do we need this class if changes were made in core that allow passing a custom `SyncAllLocationsService` to `SyncAllLocationsServiceJob` ?

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 5 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:171]: As discussed, after enabling `SyncAllLocationsServiceJob` you should be able to get the immediate parent hierarchy from the `LocationRepository`.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/job/RDTSyncAllLocationsServiceJob.java (hunk 9 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/job/RDTSyncAllLocationsServiceJob.java:14]: yes, I'll remove that

File: rdt-open-reader/src/covidIndonesia/assets/json.form/sample-collection-form.json [hunk 10]
```diff
@@ -478,7 +466,8 @@
         "v_required": {
           "value": "false",
           "err": "{{sample-collection-form-original.step1.other_health_facility_type_name.v_required.err}}"
-        }
+        },
+        "values": []
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/assets/json.form/sample-collection-form.json:470]: This can be removed.

File: rdt-open-reader/src/covidIndonesia/assets/json.form/rdt-capture-form.json [hunk 11]
```diff
@@ -157,23 +157,18 @@
...
         "openmrs_entity_id": "",
-        "type": "edit_text",
+        "type": "spinner",
+        "hint": "{{rdt-capture-form-original.step4.health_facility_name.hint}}",
         "v_required": {
           "value": "false",
           "err": "{{rdt-capture-form-original.step4.health_facility_name.v_required.err}}"
-        }
+        },
+        "values": []
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/assets/json.form/rdt-capture-form.json:171]: This can be removed.

File: rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form.json [hunk 12]
```diff
@@ -88,22 +88,18 @@
...
         "openmrs_entity_id": "",
-        "type": "edit_text",
+        "type": "spinner",
+        "hint": "{{patient-diagnostics-form-original.step1.facility_name.hint}}",
         "v_required": {
           "value": "false",
           "err": "{{patient-diagnostics-form-original.step1.facility_name.v_required.err}}"
-        }
+        },
+        "values": []
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/assets/json.form/patient-diagnostics-form.json:102]: This can be removed.

File: rdt-open-reader/src/main/AndroidManifest.xml [hunk 13]
```diff
@@ -51,6 +51,7 @@
         <service android:name="org.smartregister.sync.intent.PullUniqueIdsIntentService" />
         <service android:name=".sync.RDTSettingsSyncIntentService" />
         <service android:name="org.smartregister.sync.intent.LocationIntentService" />
+        <service android:name=".services.RDTSyncAllLocationsIntentService" />
```
Comment (vincent-karuri) [inline rdt-open-reader/src/main/AndroidManifest.xml:54]: This can be removed.

**Comment by vincent-karuri** [comment]
The sample collection form does not show the location dropdown.

**Comment by vincent-karuri** [comment]
Also, update client core version to `2.2.6-SNAPSHOT`.

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java [hunk 14]
```diff
@@ -19,5 +20,10 @@
     public static synchronized void loadLibs (Context context) {
 
     }
+
+    @Implementation
+    protected Cursor rawQueryWithFactory(SQLiteDatabase.CursorFactory cursorFactory, String sql, String[] selectionArgs, String editTable) {
```
Comment (vincent-karuri) [inline rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java:25]: Why is this needed?

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/RDTRepositoryShadow.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+import net.sqlcipher.database.SQLiteDatabase;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.shadow.api.Shadow;
+
+import io.ona.rdt.repository.RDTRepository;
+
+@Implements(RDTRepository.class)
+public class RDTRepositoryShadow extends Shadow {
```
Comment (vincent-karuri) [inline rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/RDTRepositoryShadow.java:12]: Not sure why this shadow is necessary?

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java (hunk 14 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java:25]: `RDTSettingsSyncIntentServiceTest > testOnHandleIntentShouldCallSuperAndInitiateImageUpload` test failed due to the modification of `RDTSyncIntentService` if i won't make shadow so it failed with the message of `Password has not been set`

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/RDTRepositoryShadow.java (hunk 15 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/RDTRepositoryShadow.java:12]: same as above

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java (hunk 14 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java:25]: Try adding https://github.com/onaio/rdt-standard/blob/07f2f1f889385f9e5bfd96da9b869c60d1e71356/rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteOpenHelperShadow.java#L43 and https://github.com/onaio/rdt-standard/blob/07f2f1f889385f9e5bfd96da9b869c60d1e71356/rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteOpenHelperShadow.java#L48 instead.

Comment (vincent-karuri) [reply rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java:25]: I think any method dealing with that error can be handled [here](https://github.com/onaio/rdt-standard/blob/687c7969d796a669383730c9bb1d6b76e0ac9b50/rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteOpenHelperShadow.java#L40). Do you know which line is throwing that error?

Comment (syedowaisali90) [reply rdt-open-reader/src/test/java/io/ona/rdt/robolectric/shadow/SQLiteDatabaseShadow.java]: yeah, now it's working fine. unused class removed

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/sync/RDTSettingsSyncIntentServiceTest.java [hunk 16]
```diff
@@ -2,26 +2,34 @@
...
...a751-35e8-4b46-9e53-3cb8fd193697\",\"serverVersion\":0,\"voided\":false,\"type\":\"Location\"},\"serverVersion\":0,\"voided\":false,\"type\":\"Location\"},\"parent\":\"1c7ba751-35e8-4b46-9e53-3cb8fd193697\"},\"4c7ba751-35e8-4b46-9e53-3cb8fd193697\":{\"id\":\"4c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"label\":\"Indonesia Location 3\",\"node\":{\"attributes\":{\"geographicLevel\":0.0},\"locationId\":\"4c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"name\":\"Indonesia Location 3\",\"parentLocation\":{\"locationId\":\"1c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"serverVersion\":0,\"voided\":false,\"type\":\"Location\"},\"serverVersion\":0,\"voided\":false,\"type\":\"Location\"},\"parent\":\"1c7ba751-35e8-4b46-9e53-3cb8fd193697\"}},\"id\":\"1c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"label\":\"Indonesia Division 1\",\"node\":{\"attributes\":{\"geographicLevel\":0.0},\"locationId\":\"1c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"name\":\"Indonesia Division 1\",\"serverVersion\":0,\"voided\":false,\"type\":\"Location\"}}},\"parentChildren\":{\"1c7ba751-35e8-4b46-9e53-3cb8fd193697\":[\"2c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"3c7ba751-35e8-4b46-9e53-3cb8fd193697\",\"4c7ba751-35e8-4b46-9e53-3cb8fd193697\"]}}}";
```
Comment (vincent-karuri) [inline rdt-open-reader/src/test/java/io/ona/rdt/robolectric/sync/RDTSettingsSyncIntentServiceTest.java:32]: You can share the `MOCK_LOCATION_TREE_JSON` declared here : https://github.com/onaio/rdt-standard/pull/568/files#diff-9d64eec28f3f99abb0d6e2adf95fd9cb9cc4e9c8065b87aa48756aece28c8af2R14

File: rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java [hunk 17]
```diff
@@ -845,6 +848,14 @@ else if (CovidConstants.FormFields.SENDER_NAME.equals(field.getString(JsonFormCo
             assertEquals(field.getString(VALUE), ANM_PREFERRED_NAME);
             numOfPopulatedFields++;
         }
+        else if (CovidConstants.FormFields.NAME_OF_HEALTH_FACILITY.equals(field.getString(JsonFormConstants.KEY))) {
```
Comment (vincent-karuri) [inline rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java:851]: Could you also add a separate test to verify `CovidConstants.FormFields.FACILITY_NAME` and `CovidConstants.FormFields.HEALTH_FACILITY_NAME` fields are pre-populated? You can create a `JSONObject` with these fields or read the forms directly from assets and verify that the pre-population logic works as expected.

File: rdt-open-reader/src/test/java/io/ona/rdt/robolectric/sync/RDTSettingsSyncIntentServiceTest.java (hunk 16 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/test/java/io/ona/rdt/robolectric/sync/RDTSettingsSyncIntentServiceTest.java:32]: fixed.

File: rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java (hunk 17 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java]: fixed.

**Review by vincent-karuri — APPROVED** [review]
